config/config.go

```diff
@@ -63,7 +63,7 @@ type Config map[string]*Module
 
 type WalkParams struct {
 	Version        int           `yaml:"version,omitempty"`
-	MaxRepetitions uint8         `yaml:"max_repetitions,omitempty"`
+	MaxRepetitions uint32        `yaml:"max_repetitions,omitempty"`
```

Comment:
FYI, before this commit, max_repetitions was stored as an uint8. If max_repetitions is set to anything greater than 127, it was encoded (ASN.1/DER) as "02 01 80", which, AFAIK, really means -128, breaking the request. 128 should be "02 02 00 80" as there is no unsigned integers for DER. I would double-check the SNMP library if it is correctly encoding unsigned values as it might break other scenarios.

I think that, with uint32, it is virtually impossible to actually use anything greater than 2^31. I didn't test this patch and I'll wait for the next release. Anyway, it would be nice to duble check this.

Reply:
This change was related to an upstream gosnmp change.

https://github.com/gosnmp/gosnmp/pull/295.